lib/spack/spack/package.py

```diff
@@ -1668,6 +1668,7 @@ def run_test(self, exe, options=[], expected=[], status=None,
     def _run_test_helper(self, runner, options, expected, status, installed,
                          purpose):
         status = [status] if not isinstance(status, list) else status
+        expected = [expected] if not isinstance(expected, list) else expected
```

Comment:
```suggestion
        expected = [expected] if isinstance(expected, six.string_types) else expected
```

That way we can use generators and other iterables seamlessly in place of a list.

Reply:
How are six.string_types not covered by the original code?

Reply:
Will also change the other options for consistency.

Reply:
Done.